File: backend/src/main/java/be/lutske/leolegacy/infrastructure/persistence/entity/RecipeEntity.java
package be.lutske.leolegacy.infrastructure.persistence.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "recipe")
public class RecipeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private String imageUrl;
    private String ingredients;
    private String instructions;
    private int viewCount;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private CategoryEntity category;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public String getIngredients() { return ingredients; }
    public void setIngredients(String ingredients) { this.ingredients = ingredients; }

    public String getInstructions() { return instructions; }
    public void setInstructions(String instructions) { this.instructions = instructions; }

    public int getViewCount() { return viewCount; }
    public void setViewCount(int viewCount) { this.viewCount = viewCount; }

    public CategoryEntity getCategory() { return category; }
    public void setCategory(CategoryEntity category) { this.category = category; }
}

File: backend/src/main/kotlin/be/lutske/leolegacy/infrastructure/persistence/entity/RecipeEntity.kt
package be.lutske.leolegacy.infrastructure.persistence.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "recipe")
class RecipeEntity : PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    lateinit var title: String

    @Column(columnDefinition = "TEXT")
    lateinit var ingredients: String

    @Column(columnDefinition = "TEXT")
    lateinit var preparation: String

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    lateinit var category: CategoryEntity

    @Column(name = "view_count")
    var viewCount: Int = 0

    @Column(name = "source")
    var source: String? = null

    @Column(name = "created_at")
    var createdAt: Instant? = null

    constructor()

    constructor(
        id: Long,
        title: String,
        ingredients: String,
        preparation: String,
        category: CategoryEntity,
        viewCount: Int = 0,
        source: String? = null,
        createdAt: Instant? = null
    ) {
        this.id = id
        this.title = title
        this.ingredients = ingredients
        this.preparation = preparation
        this.category = category
        this.viewCount = viewCount
        this.source = source
        this.createdAt = createdAt
    }
}
